drivers/flash/flash_stm32h7x.c

```diff
@@ -225,15 +302,19 @@ static int write_ndwords(const struct device *dev,
 	uint32_t tmp;
 	int rc;
 	int i;
-	uint32_t sector = get_sector(offset);
+	struct flash_stm32_sector sector = get_sector(offset);
 
-	if (sector < FLASH_SECTOR_TOTAL) {
+	switch (sector.bank) {
+	case 1:
 		cr = &regs->CR1;
+		break;
 #ifdef DUAL_BANK
```

Comment:
Same here. Put it in the get_sector() function.

Reply:
Done